psamm_import/importer.py

```diff
@@ -123,6 +123,39 @@ def _decimal_representer(dumper, data):
     return dumper.represent_scalar('tag:yaml.org,2002:float', value)
 
 
+def get_default_compartment(model):
+    """Return what the default compartment should be set to.
+
+    If some compounds have no compartment, unique compartment
+    name is returned to avoid collisions.
+    """
+    default_compartment = 'c'
+    default_key = set()
+    suffix = 0
+    for reaction_id, reaction in iteritems(model.reactions):
+        if 'equation' not in reaction.properties:
+            continue
+
+        equation = reaction.properties['equation']
+        for compound, _ in equation.compounds:
+            default_key.add(compound.compartment)
+
+    while True:
+        if None in default_key and default_compartment in default_key:
+            suffix += 1
+            new_key = '{}_{}'.format(default_compartment, suffix)
+            if new_key not in default_key:
+                default_compartment = new_key
+                logger.warning(
+                    "Compound(s) found without compartment," +
+                    "default compartment is" + default_compartment + ".")
+                break
+            suffix += 1
+        else:
+            break
```

Comment:
Note that you are incrementing suffix twice so you are going `c_1`, `c_3`, `c_5`, etc. Use `format()` instead of text concatenation with `+`. Also, you're missing a space in the warning string. I think the loop would be less complicated if the first `if` statement is outside (before) the loop and the warning is outside the loop (after). Initialize variables where they are used, i.e. move `suffix = 0` close to the loop. You can also get rid of the `else: break`:

``` python
if None in default_key and default_compartment in default_key:
    suffix = 1
    while True:
        new_key = '{}_{}'.format(default_compartment, suffix)
        if new_key not in default_key:
            default_compartment = new_key
            break
        suffix += 1

if None in default_key:
    logger.warning('Compound(s) found without compartment, default compartment is {}'.format(
        default_compartment))
```